R/output-batches.R

```diff
@@ -19,17 +19,68 @@
 #' - `<lazyframe>$sink_batches()` returns `NULL` invisibly.
 #' - `<lazyframe>$lazy_sink_batches()` returns a new [LazyFrame].
 #' @examples
+#' # One usecase for this function is to export larger-than-RAM data to file
+#' # formats for which polars doesn't provide a writer out of the box.
+#' # The example below writes a LazyFrame by batches to a CSV file for the
+#' # sake of the example, but one could replace `write.csv()` by
+#' # `haven::write_dta()`, `saveRDS()`, or other functions.
 #' lf <- as_polars_lf(mtcars)
-#' lf$sink_batches(\(df) print(df), chunk_size = 10)
 #'
-#' # Early stopping by returning TRUE from the function
-#' lf$sort("cyl")$sink_batches(
+#' output_dir <- withr::local_tempdir()
+#' output_dir_idx <- 1
+#'
+#' lf$sink_batches(
+#'   \(df) {
+#'      # `df` is one chunk of the original LazyFrame.
+#'      dest <- paste0(output_dir, "/file_", output_dir_idx, ".csv")
+#'      cat(sprintf("Writing %s rows to %s\n", nrow(df), dest))
+#'      write.csv(as.data.frame(df), dest)
```

Comment:
Weirdly, this also works completely fine with `write.csv(df, dest)` while `df` is a `DataFrame`. I don't really know why but `haven::write_dta(df)` errors for instance.